src/com/jwetherell/algorithms/mathematics/Division.java

```diff
@@ -11,6 +16,10 @@ public static final long divisionUsingLoop(int a, int b) {
         int absA = Math.abs(a);
         int absB = Math.abs(b);
 
+        if (b == 0) {
```

Comment:
The method will throw an exception by ensuring the value of b before the result "long" is calculated avoiding unexpected results.